packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -160,24 +162,32 @@ const AccountRecoveryConfirmKey = (_: RouteComponentProps) => {
       fetchedResetToken,
       getRecoveryBundleAndNavigate,
       setLinkStatus,
-      ftlMsgResolver,
+      errorInvalidRecoveryKey,
     ]
   );
 
-  const onSubmit = ({ recoveryKey, token, code, email, uid }: SubmitData) => {
+  const invalidRecoveryKeyLength = (localizedError: string) => {
+    setRecoveryKeyErrorText(localizedError);
+    setIsLoading(false);
+    logViewEvent('flow', `${viewName}.fail`, REACT_ENTRYPOINT);
+  };
+
+  const onSubmit = (submitData: SubmitData) => {
     setIsLoading(true);
     logViewEvent('flow', `${viewName}.submit`, REACT_ENTRYPOINT);
-    if (recoveryKey === '') {
-      const errorEmptyRecoveryKeyInput = ftlMsgResolver.getMsg(
-        'account-recovery-confirm-key-empty-input-error',
-        'Account recovery key required'
+
+    if (submitData.recoveryKey === '') {
+      invalidRecoveryKeyLength(
+        ftlMsgResolver.getMsg(
+          'account-recovery-confirm-key-empty-input-error',
+          'Account recovery key required'
+        )
       );
-      setRecoveryKeyErrorText(errorEmptyRecoveryKeyInput);
-      setIsLoading(false);
-      return;
+    } else if (submitData.recoveryKey.length !== 32) {
```

Comment:
While this does work, I think it might be better to do similar formatting that is done on content server (replace spaces with '').

https://github.com/mozilla/fxa/blob/f28e95071bedb319e19932e02811a6f46fa4a1f3/packages/fxa-content-server/app/scripts/views/elements/recovery-key-input.js#L21

Reply:
You can also validate that this is indeed a base32 string too, https://github.com/mozilla/fxa/blob/506227452748bbd9d6e33fb9c3f73f50180c3f79/packages/fxa-content-server/app/scripts/lib/validate.js#L188.

Reply:
We do that [here](https://github.com/mozilla/fxa/blob/e0be7066118e561f6b256dcef93068bf68793bd4/packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx#L236), the `submitData.recoveryKey.length !== 32` check occurs after the replace.

Reply:
I added this, however I noticed the existing check in content-server didn't reject `L`s, and it wasn't case insensitive (I think case insensitivity is fine, in case a user printed their recovery code or wrote it down and typed it in, only to be met with an "invalid recovery key" message). I accounted for this, and added a few quick tests too.